src/daemon/common.c

```diff
@@ -361,13 +361,14 @@ int strjoin (char *buffer, size_t buffer_size,
 			avail -= sep_len;
 		}
 
-		field_len = strlen (fields[i]);
-		if (avail < field_len)
+		if (avail == 0)
 			return (-1);
 
-		memcpy (ptr, fields[i], field_len);
+		field_len = strlen (fields[i]);
+
+		memcpy (ptr, fields[i], ((avail < field_len) ? avail : field_len));
 		ptr += field_len;
```

Comment:
Figured if avail was used instead of fieldlen, we've run out of space
anyway (we're either done concating or avail will be 0 and loop will
return(-1) when attempting the next string just as before).  Doesn't hurt
to improve the completeness and readability, though...

--John

On Thu, Jul 14, 2016, 01:21 Peter Portante <notifications@github.com> wrote:

> In src/daemon/common.c
> <https://github.com/collectd/collectd/pull/1792#discussion_r70749900>:
>
> >  			return (-1);
> >
> > -		memcpy (ptr, fields[i], field_len);
> > +		field_len = strlen (fields[i]);
> > +
> > +		memcpy (ptr, fields[i], ((avail < field_len) ? avail : field_len));
> >  		ptr += field_len;
>
> Should ptr be incremented by the length actually copied?
>
> So perhaps we could have:
>
> field_len = strlen (fields[i]);
> copy_len = ((avail < field_len) ? avail : field_len);
> memcpy (ptr, fields[i], copy_len);
> ptr += copy_len;
> avail -= copy_len;
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/collectd/collectd/pull/1792/files/b344d9252ef3322fe71c982627e0694a89f31b55#r70749900>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGhlE-DsdmTwWjL3TJIXhVWNpzQGu8ckks5qVcdrgaJpZM4JL2nS>
> .
>